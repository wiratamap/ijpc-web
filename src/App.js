import React from 'react';
import 'bootstrap/dist/css/bootstrap.css';
import './index.css';

import VacancyContainer from './Containers/Vacancy/VacancyContainer';

const App = () => (
  <div className="container-fluid min-vh-100 bg-main">
    <VacancyContainer />
  </div>
);

export default App;
