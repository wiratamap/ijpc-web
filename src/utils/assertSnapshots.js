import React from 'react';
import renderer from 'react-test-renderer';

/**
 * @param Component - (Component) Component that needs to be test.
 * @param testConfiguration - (object) object of props and test description
 */
const assertSnapshot = (Component, testConfiguration) => {
  const { props, description } = testConfiguration;

  describe('#render', () => {
    it(description, () => {
      const component = renderer.create(<Component {...props} />).toJSON();

      expect(component).toMatchSnapshot();
    });
  });
};

const assertSnapshots = (Component, testConfigurations) => {
  testConfigurations.forEach((testConfiguration) => assertSnapshot(Component, testConfiguration));
};

export default assertSnapshots;
