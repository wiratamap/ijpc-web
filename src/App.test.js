import App from './App';
import assertSnapshots from './utils/assertSnapshots';

jest.mock('./Containers/Vacancy/VacancyContainer', () => 'VacancyContainer');

describe('App', () => {
  const configurations = [
    {
      description: 'should render vacancy list component with pagination component'
    }
  ];

  assertSnapshots(App, configurations);
});
