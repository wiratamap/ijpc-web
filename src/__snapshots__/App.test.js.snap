// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App #render should render vacancy list component with pagination component 1`] = `
<div
  className="container-fluid min-vh-100 bg-main"
>
  <VacancyContainer />
</div>
`;
